docs/design/2022-11-17-ttl-table.md

```diff
@@ -0,0 +1,327 @@
+# Proposal: Support TTL Table
+- Author(s): [lcwangchao](https://github.com/lcwangchao)
+- Tracking Issue: https://github.com/pingcap/tidb/issues/39262
+
+## Table of Contents
+
+<!-- TOC -->
+* [Proposal: Support TTL Table](#proposal--support-ttl-table)
+  * [Table of Contents](#table-of-contents)
+  * [Introduction](#introduction)
+  * [Detailed Design](#detailed-design)
+    * [Syntax](#syntax)
+      * [Create TTL Table](#create-ttl-table)
+      * [Alter a Table with TTL](#alter-a-table-with-ttl)
+      * [Alter to a non-TTL Table](#alter-to-a-non-ttl-table)
+      * [Constraints](#constraints)
+    * [TTL Job Management](#ttl-job-management)
+    * [TTL Job Details](#ttl-job-details)
+      * [Scan Tasks](#scan-tasks)
+      * [Delete Tasks](#delete-tasks)
+    * [New System Variables](#new-system-variables)
+    * [New Metrics](#new-metrics)
+  * [Known Issues](#known-issues)
+  * [Future Works](#future-works)
+  * [Alternative Solutions](#alternative-solutions)
+<!-- TOC -->
+
+## Introduction
+
+The rows in a TTL table will be deleted automatically when they are expired. It is useful for some scenes, for example, delete the expired verification codes. A TTL table will have a column with the type DATE/DATETIME/TIMESTAMP which will be compared with the current time, if the interval between them exceeds some threshold, the corresponding row will be deleted.
```

Comment:
Please specify that DATE/DATETIME is parsed as they are in SYSTEM time zone. TIMESTAMP should internally be stored as UTC, but it may have some corner cases in comparison when Day Light Saving Time transitions the clock backwards.

Reply:
Well, I think we should parse the DATE/DATETIME according to the sys var `time_zone` whose default value is "SYSTEM" . Day Light Saving Time is a problem, I'll look at it later. Thanks for this suggestion.